docs/src/adr/020adr-prioritization.md

```diff
@@ -0,0 +1,204 @@
+- Medium **`e1`**: Can be completed within 3 days
+- Difficult **`e2`**: Will take 5 or more days
+
+Ideally, as soon as we recognized that an actively planned ticket will take more
+than 5 days of focused work, we would factor it into smaller tickets, allowing
+us to avoid perpetually prolonged monster tickets. But some people may prefer to
+track "major projects" with a single issue rather than a milestone, and this
+doesn't pose a significant problem.
+
+##### Impact
+
+Impact is scored based on considering impact in the follow 4 domains:
+
+1. _User / customer_: The benefit of the results of a work item to our users and
+   customers
+2. _Mission_: Furtherance of our organization's mission and objectives
+3. _Invention_: Advancement of state of the art in formal verification and
+   specification
+4. _Development_: Improvements to our development capacities and bandwidth
+   (which supports advancing the other three factors).
+
+- High **`i0`**:
+  - User / customer: Unblocks critical work
+  - Mission: Advances critical organizational objectives
+  - Invention: Opens up novel verification and specifications abilities
+  - Development: Saves > 3 hours per week
+- Medium **`i1`**:
+  - User / customer: Unblocks non-critical work
+  - Mission: Advances non-critical organizational objectives
+  - Invention: Makes incremental improvement to verification and specification
+    abilities
+  - Development: Saves between 1 to 3 hours per week
+- Low **`i2`**:
+  - User / customer: Improves functionality, but an easy workaround exists
+  - Mission: No significant advance of organizational objectives
+  - Invention: No significant improvement to verification and specification abilities
+  - Development: Saves < 1 hours per week
+
+### Prioritization and evaluation
+
+The initial priority of a requirement should be established at the time work is
+agreed upon, including when setting plans quarterly and when triaging unplanned
+work.
+
+When unplanned work is introduced, the stakeholders involved should determine
+its score (using informal language, when needed), and this should be used to
+decided whether it is worth interrupting any ongoing or planned work.
+
+When deciding which work item to take on next, we should favor work that is
+nearest to scoring minimum effort and maximum impact: (`e0`, `i0`). Ties should
+be resolved based on the worker's inclination or discussions within other
+stakeholders.
+
+The priority of work should be re-evaluated as the situation changes. E.g.:
+
+- When goals change, then impact should be reconsidered.
+- If we discover work was incorrectly scoped, then the effort should be
+  reevaluted.
+
+We can generate views into the 4 quadrants via filters in our project board, or
+any other tooling or visualization we find suitable. When reviewing work
+in progress or queued up next at our weekly meetings, we should always be
+sure that the highest priority work on the board is being addressed.
```

Comment:
My understanding from the reading was that the effort vs. impact graph falls into 4 quadrants: this is presented in every article I read on the approach, and in the table presented here.

I'm fine with using whatever visualization methods find effective, but I don't know what tooling for that we can incorporate into our workflow without imposing a large overhead.

The three articles I read on the APM methodology (including the one you linked originally) advise listing all the items to work on, then scoring them from 1 to 10, then using the scores to graph them. None of them mention scoring things by placing things in a graph, as far as I recall (bu maybe I missed that somewhere?). So I tried to adapt that, but with lower overhead on the scoring.

As I said, I'm open to integrating other tooling etc., but that will need concrete input to actualize I think.

Reply:
> My understanding from the reading was that the effort vs. impact graph falls into 4 quadrants: this is presented in every article I read on the approach, and in the table presented here.

Yes, but they are continuous graph quadrants (as opposed to a table), like this image from one of the linked references:
![image](https://user-images.githubusercontent.com/18356186/170271136-cb6d0082-f79e-435d-90e8-2a183934facf.png)

So my confusion was when you mentioned using the project board to visualize it. Unfortunately I also don't now of any tooling to plot this automatically.

Anyway, my point here was just to make sure I understood correctly and raise a concern about limitations on visualization that I don't know how to solve. This is just something to keep in mind if we face difficulties when trying to assess priority of new items over time.

Reply:
Yeah, I 'm suggesting really fudging the scoring the part by reducing to 3 values. I don't know if that's too restrictive.

For use of the project board, I just figured we can create 4 tables with filters for the combination of labels that would put something in a quadrant. Then we'll have to sort or just eyeball any priority difference within the quadrant. Not very elegant, and if the visualization is really essential, maybe it would be at cross purposes?

Reply:
Makes sense! Let's see how much overhead the lack of visualization will have on our decision process (on what to do next). I hope we can find enough workarounds on the project board, like the one you're suggesting, and that they are enough to help us reason about the dimensions.